### ISSUE TITLE
Use facenet 20170512-110547.pb model in Android meet "No OpKernel was registered to support Op 'Switch' " error 

### ISSUE THREAD

**Issue by banderuilq** [davidsandberg/facenet#348]
Use facenet 20170512-110547.pb model in Android meet error .That is"No OpKernel was registered to support Op 'Switch' with these attrs". The error as below:
06-22 20:02:37.725 16360-16383/com.tzutalin.dlibtest E/TensorFlowInferenceInterface: Failed to run TensorFlow inference with inputs:[input], outputs:[embeddings]

                                                                                     --------- beginning of crash
06-22 20:02:37.764 16360-16383/com.tzutalin.dlibtest E/AndroidRuntime: FATAL EXCEPTION: pool-1-thread-1
                                                                       Process: com.tzutalin.dlibtest, PID: 16360
                                                                       java.lang.IllegalArgumentException: No OpKernel was registered to support Op 'Switch' with these attrs.  Registered devices: [CPU], Registered kernels:
                                                                         device='GPU'; T in [DT_STRING]
                                                                         device='GPU'; T in [DT_BOOL]
                                                                         device='GPU'; T in [DT_INT32]
                                                                         device='GPU'; T in [DT_FLOAT]
                                                                         device='CPU'; T in [DT_FLOAT]
                                                                         device='CPU'; T in [DT_INT32]

                                                                            [[Node: InceptionResnetV1/Conv2d_1a_3x3/BatchNorm/cond/Switch = Switch[T=DT_BOOL](phase_train, phase_train)]]
                                                                           at org.tensorflow.Session.run(Native Method)
                                                                           at org.tensorflow.Session.access$100(Session.java:48)
                                                                           at org.tensorflow.Session$Runner.runHelper(Session.java:295)
                                                                           at org.tensorflow.Session$Runner.run(Session.java:245)
                                                                           at org.tensorflow.contrib.android.TensorFlowInferenceInterface.run(TensorFlowInferenceInterface.java:142)

**Comment by banderuilq** [comment]
I write the java code in Android as below:
 public  void getImageData(ArrayList<String> imagePath, Context context){
      Log.d("Facenet"," getImageData in");
      if (!imagePath.isEmpty())
      {

         float[] iamgesPixels =  new float[imagePath.size()* 160 *160 * 3];
          Log.d("Facenet"," getImageData in not empty iamgesPixels length = "+iamgesPixels.length);
          float[] outputs = new float[1024];
         for (int i = 0; i < imagePath.size(); ++i)
         {
             Log.d("Facenet","imagePath = "+imagePath + " i = "+ i);
            Bitmap bitmap = BitmapFactory.decodeFile(imagePath.get(i));
            float[] imageArray = bitmapToFloatArray(bitmap);
             System.arraycopy(imageArray,0,iamgesPixels,i*160 *160 * 3,imageArray.length);
         }
          inferenceInterface = new TensorFlowInferenceInterface(context.getAssets(), MODEL_FILE);
          inferenceInterface.feed(INPUT_NODE, iamgesPixels, 921600);
          inferenceInterface.run(new String[]{OUTPUT_NODE}, false);
          inferenceInterface.fetch(OUTPUT_NODE, outputs);
          Log.d("Facenet","outputs.length = "+ outputs.length);
      }
   }

**Comment by banderuilq** [comment]
If you have fix the issue “No OpKernel was registered to support Op 'Switch' with these attrs".Please help me，Thanks.

**Comment by aigeek** [comment]
Hi, do you know how to use this project in andorid ? Is there some tutorial? Tks! @banderuilq

**Comment by apollo-time** [comment]
#238
You must freeze graph without any file input nodes.

**Comment by lili25** [comment]
@apollo-time  Could  you share how to freeze the graph without  input node "phase_train" ?
thanks.

**Comment by Dicksonchin93** [comment]
@apollo-time Hi, Do you have code to calculate the L2 distance between the embeddings in Java? I am getting varying distance that don't really make sense with my implemention of the distance calculation from your facenet.pb file in android. Appreciate if you could share if you have.

**Comment by mixml** [comment]
I have meet this problem, How to fix it?

**Comment by Hellotianz** [comment]
how to use faceNet in Android？ Can you share with me

**Comment by vitrol4** [comment]
See here guys: https://github.com/davidsandberg/facenet/issues/238

And use this .pb model: https://github.com/apollo-time/facenet/raw/715d5cfbd2eef70e83f55ff9b697d13173ee06de/model/resnet/facenet.pb
